### ISSUE TITLE
Support non-capturing group in native language.

### ISSUE THREAD

**Issue by fdutton** [DmitrySoshnikov/syntax#30]
Some languages like PHP do not adhere to the POSIX grammar. Since lex-rule.js instantiates a field using JavaScript's RegExp, it implicitly expects the expression to conform to the POSIX grammar and will raise a "Invalid regular expression" exception if it does not.

Example:
PCRE: ```(?<=[:space:])[Aa][Nn][Dd](?=[:space:])```
POSIX: ```(?:[:space:])[Aa][Nn][Dd](?=[:space:])```

Notes:
1. The upper- and lower-case variants exist because I was unable to determine how to make the parser case insensitive.
2. The reason for the non-capturing groups is because the parser matched "and" in "android".

**Comment by DmitrySoshnikov** [comment]
@fdutton, thanks for reporting the issues, all are good points.

- For the case-insensitivity, I opened a separate issue #31.

- For the matching "and" in "android": it should be possible to fix it using word boundary: `\b`:

```php
// PHP

preg_match('/and/', 'android', $m); // 1
preg_match('/\band\b/', 'android', $m); // 0
```

- And for different regexp syntax, like `(?<=[:space:])[Aa][Nn][Dd](?=[:space:])` (which is not supported by JS) -- yes we should fix it. This should be an easy fix: instead of [creating the matcher](https://github.com/DmitrySoshnikov/syntax/blob/master/src/grammar/lex-rule.js#L32) directly inside the constructor, we can introduce `_buildMatcher` function, and create it in `try-catch` only if the target language is JS. Similarly how we do this in [_buildHandler](https://github.com/DmitrySoshnikov/syntax/blob/master/src/grammar/lex-rule.js#L115-L117) skipping it for other languages. This should be fine, since plugin code generators do not use it, and use [rawMatcher](https://github.com/DmitrySoshnikov/syntax/blob/master/src/plugins/php/php-parser-generator-trait.js#L85) instead.

I'll try to take a look at this issue a bit later, however, if you'd like to send a small PR for it, I'll appreciate your help, and will be glad to review the diff.

**Comment by DmitrySoshnikov** [comment]
OK, this should be fixed in `0.0.57`, and this should work:

```
(?:[[:space:]])[Aa][Nn][Dd](?=[[:space:]])
```

Notice though, the first example with _lookbehind assertion_ won't work, since tokenizer applies all rules starting from the beginning of a string (i.e. adding `^` at the beginning of all regexes).

```
^(?<=[[:space:]])[Aa][Nn][Dd](?=[[:space:]])
```

("lookbehind" doesn't make sense for at the beginning of a string)

Still, using `\b` would fit better for your use-case.

For the case-insensitive follow the #31.

**Comment by DmitrySoshnikov** [comment]
For the _lookbehind assertions_ I opened #32. This should be written explicitly as (see `^`):

```
'/(?<=^[[:space:]])A(?=[[:space:]])/'
```

**Comment by DmitrySoshnikov** [comment]
Re: case-insensitive: fixed by https://github.com/DmitrySoshnikov/syntax/commit/c1442bba16f6e8b52219af49d5abb9582cb6b288, and is available since `0.0.57`. See docs [here](https://www.npmjs.com/package/syntax-cli#case-insensitive-match).
